graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -216,6 +217,7 @@ public DateTime getTimestamp() {
 
         obj.put(FIELD_MESSAGE, getMessage());
         obj.put(FIELD_SOURCE, getSource());
+        obj.put(FIELD_STREAMS, getStreamIds());
```

Comment:
This doesn't fix the issue with the archive restore.

In the restore case the stream IDs are in the `fields` map but the `streams` set is empty and thus the stream IDs in `fields` will be cleared.